corehq/apps/smsbillables/management/commands/retrobill_india_unicel_03012016.py

```diff
@@ -0,0 +1,106 @@
+                domain=unique_b.domain,
+            )
+            total_affected = all_affected_billables.count()
+            if total_affected > 0:
+                print(
+                    "\nUpdating {total_affected} billables for"
+                    " domain {domain}".format(
+                        domain=unique_b.domain,
+                        total_affected=total_affected
+                    )
+                )
+                stdout.write(">> BILLABLES: ")
+            total_diff = Decimal('0.0')
+            for billable in all_affected_billables.all():
+                stdout.write('.')
+                updated_billable = self._get_updated_billable(billable, inr)
+
+                old_gateway_cost = (
+                    billable.gateway_fee.amount /
+                    billable.gateway_fee_conversion_rate
+                )
+                new_gateway_cost = (
+                    updated_billable.gateway_fee.amount /
+                    updated_billable.gateway_fee_conversion_rate
+                )
+
+                difference = old_gateway_cost - new_gateway_cost
+                total_diff += difference * Decimal('1.0000')
+                total_diff += difference * Decimal('1.0000')
+            stdout.flush()
+            if total_diff > Decimal('0.0'):
+                print(
+                    "\n!! >>> FOUND difference of {diff}, "
+                    "applying General Credits to domain {domain}".format(
+                        diff=round(total_diff, 4),
+                        domain=unique_b.domain,
+                    )
+                )
+                try:
+                    affected_account = BillingAccount.get_account_by_domain(unique_b.domain)
+                    CreditLine.add_credit(
+                        total_diff,
+                        affected_account,
+                        note="Automated re-calc for UNICEL SMS Fees due to incorrect"
+                             "conversion rate",
+                        reason=CreditAdjustmentReason.MANUAL,
+                    )
+                    for b in all_affected_billables.all():
+                        b.is_valid = False
+                        b.save()
+                except Exception as e:
+                    print("Could not add credits to {domain} due to {error}".format(
+                        domain=unique_b.domain,
+                        error=e
+                    ))
+
+    def _get_updated_billable(self, old_billable, updated_currency):
+        new_billable = SmsBillable(
+            log_id=old_billable.log_id,
+            phone_number=old_billable.phone_number,
+            direction=old_billable.direction,
+            date_sent=old_billable.date_sent,
+            domain=old_billable.domain,
+            multipart_count=old_billable.multipart_count
+        )
+        new_billable.gateway_fee = old_billable.gateway_fee
+        new_billable.gateway_fee_conversion_rate = updated_currency.rate_to_default
```

Comment:
another approach would be to update ```gateway_fee_conversion_rate``` for the affected billables, to save the work/risk of error in creating new billables and invalidating old ones.